### PR TITLE
add seldon

### PR DESCRIPTION
**Pull request by cliveseldon**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/36)

### CODE REVIEW

File: seldon/base/seldon-core-seldon-cluster-manager-deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      app.kubernetes.io/component: seldon-core-operator
+      app.kubernetes.io/name: seldon-core
+      component: seldon-core
+  template:
+    metadata:
+      labels:
+        app: seldon-cluster-manager-server
+        app.kubernetes.io/component: seldon-core-operator
+        app.kubernetes.io/name: seldon-core
+        component: seldon-core
```
Comment (hougangliu) [inline seldon/base/seldon-core-seldon-cluster-manager-deploy.yaml]: you can remove `matchLabels` and `labels` here since `commonLabels` has done it

Comment (cliveseldon) [reply seldon/base/seldon-core-seldon-cluster-manager-deploy.yaml]: done

**Comment by cliveseldon** [comment]
@kkasravi Updated for latest Seldon and added application.yaml

File: seldon/seldon-core-operator/base/application.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+apiVersion: app.k8s.io/v1beta1
```
Comment (kkasravi) [inline seldon/seldon-core-operator/base/application.yaml]: this should be moved to seldon/seldon-core-operator/overlays/application/

jupyter/notebook-controller has an example including setting commonLabels in the kustomization.yaml.
All labels and matchLabels should be deleted and replaced by the commonLabels in seldon/seldon-core-operator/overlays/application/kustomization.yaml

File: seldon/seldon-core-operator/base/seldon-operator-controller-manager-service-svc.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    control-plane: controller-manager
+    controller-tools.k8s.io: "1.0"
+  name: seldon-operator-controller-manager-service
+  namespace: kubeflow
```
Comment (kkasravi) [inline seldon/seldon-core-operator/base/seldon-operator-controller-manager-service-svc.yaml]: you can delete `namespace: kubeflow` in all resources, it's set by kfctl

Comment (cliveseldon) [reply seldon/seldon-core-operator/base/seldon-operator-controller-manager-service-svc.yaml]: Done except for cluster role binding

**Comment by cliveseldon** [comment]
Do I need to add a go test file for seldon?
I see when you run the `make generate` in the tests folder it deletes the go files. After tests I assume we are meant to restore them?

**Comment by jlewi** [comment]
@hougangliu @kkasravi is OOO can you review this please?

**Comment by hougangliu** [comment]
@jlewi sure, I will take a look later today

File: seldon/values.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+ambassador:
```
Comment (hougangliu) [inline seldon/values.yaml]: is this file must be maintained here?

File: seldon/seldon-core-operator/base/service.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  labels:
+    control-plane: controller-manager
+    controller-tools.k8s.io: "1.0"
+  name: seldon-operator-controller-manager-service
+spec:
+  ports:
+  - port: 443
+  selector:
+    control-plane: controller-manager
```
Comment (hougangliu) [inline seldon/seldon-core-operator/base/service.yaml]: selector here is too general, it looks like no Seldon-related. why not "seldon-operator-controller-manager"?
also maybe you can add commonLabels in base/kustomization.yaml

File: seldon/values.yaml (hunk 4 above)
Comment (cliveseldon) [reply seldon/values.yaml]: Its a file for settings to convert the core Seldon helm chart to kustomize so future updates can use it.

File: seldon/seldon-core-operator/base/service.yaml (hunk 5 above)
Comment (cliveseldon) [reply seldon/seldon-core-operator/base/service.yaml]: I changed selector. Common labels are in overlays/application as suggested by @kkasravi

**Comment by hougangliu** [comment]
/lgtm
/approve
thanks

**Pull request merged**